hyperspy/drawing/utils.py

```diff
@@ -878,135 +911,206 @@ def check_list_length(arg, arg_name):
 
     # Replot: create a list to store references to the images
     replot_ims = []
+    
+    #Below is for overlayed images
+    if overlay:
+        
+        #Check if images all have same scale and therefore can be overlayed.
+        for im in images:
+            if (im.axes_manager[0].scale != 
+                images[0].axes_manager[0].scale):
+                raise ValueError("Images are not the same scale and so should"
+                                 "not be overlayed.")
+        
+        original_cmap = plt.rcParams['image.cmap']
+        plt.rcParams['image.cmap'] = 'gray'
+        import matplotlib.patches as mpatches
+        ax = f.add_subplot(1, 1, 1)
+        patches = []
+        
+        #If no colors are selected use BASE_COLORS
+        if colors == []:
+            for i in range(len(images)):
+                colors.append(list(BASE_COLORS)[i])
```

Comment:
Define it in the definition of the function: less code for us and more explicit for the user.